### ISSUE TITLE
unable to assemble uncorrected reads

### ISSUE THREAD

**Issue by ShuChen1986** [marbl/canu#865]
Hi,
I am using canu 1.7 to assemble pacbio reads.
This is the command lines I used:

```
perl /home/user/Music/canu-1.7/Linux-amd64/bin/canu \
-assemble \
-p stylo-lordec \
-d /home/user/Documents/Result/0result_canu/canu_2018-04-10/output \
genomeSize=1.5G \
maxThreads=11 \
maxMemory=100000000000 \
-pacbio-raw \
/home/user/Documents/Stylosanthes/lordec/stylo_pacbio_combined.trimmed.fq
```

This is the error info I got from running the command lines:
**ABORT:
ABORT: Canu 1.7
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   unable to assemble uncorrected reads.**

This issue did not occur when I used the same command lines for canu 1.6.
I am assembling the reads as -pacbio-raw, the program should correct the reads before assembling, why did it remind me it was unable to assemble uncorrected reads?
Your help is really appreciated. Thank you so much in advance.

Shu

**Comment by skoren** [comment]
This is a new check added to 1.7 to prevent common errors. By specifying -assemble you're asking canu to not correct or trim the data just assemble as is. This is not likely to work for raw input data which is why Canu is exiting. If that's what you want you should be providing corrected reads to Canu. If you want Canu to correct/trim/assemble the data then don't specify the -assemble flag.

**Comment by ShuChen1986** [comment]
Thank you so much! I will run canu without the assemble flag.

**Comment by DongnaMa** [comment]
Hi，I am using canu 1.7.1 to assemble pacbio reads. I used: canu -assemble -s spec.txt -p Pe -d Pe.pacbio  genomeSize=1.27g -pacbio-corrected ../1_data/filtered_subreads.fasta 
 I got a error:

-- In 'Pe.gkpStore', found PacBio reads:
--   Raw:        18046215
--   Corrected:  0
--   Trimmed:    0
--
-- Generating assembly 'Pe' in '/home/stu_madongna/project/5_baixiangguo/2_Assemble/Pe.pacbio'
--
-- Parameters:
--
--  genomeSize        1270000000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.2400 ( 24.00%)
--    obtOvlErrorRate 0.0450 (  4.50%)
--    utgOvlErrorRate 0.0450 (  4.50%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.3000 ( 30.00%)
--    obtErrorRate    0.0450 (  4.50%)
--    utgErrorRate    0.0450 (  4.50%)
--    cnsErrorRate    0.0750 (  7.50%)
--
--
-- BEGIN ASSEMBLY
--
-- Unable to assemble uncorrected reads.

ABORT:
ABORT: Canu 1.7.1
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   unable to assemble uncorrected reads.
ABORT:

canu you give me some advance?
Thanks
Molly

**Comment by DongnaMa** [comment]
Hi,I used canu 1.6 to assemble pacbio reads, but Task bat can't run and then I used  1.7.1 version.This is the command lines I used:
canu -assemble -s spec.txt -p Pe -d Pe.pacbio  genomeSize=1.27g -pacbio-corrected ../1_data/filtered_subreads.fasta
I got a error:

-- In 'Pe.gkpStore', found PacBio reads:
--   Raw:        18046215
--   Corrected:  0
--   Trimmed:    0
--
-- Generating assembly 'Pe' in 'Pe.pacbio'
--
-- Parameters:
--
--  genomeSize        1270000000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.2400 ( 24.00%)
--    obtOvlErrorRate 0.0450 (  4.50%)
--    utgOvlErrorRate 0.0450 (  4.50%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.3000 ( 30.00%)
--    obtErrorRate    0.0450 (  4.50%)
--    utgErrorRate    0.0450 (  4.50%)
--    cnsErrorRate    0.0750 (  7.50%)
--
--
-- BEGIN ASSEMBLY
--
-- Unable to assemble uncorrected reads.

ABORT:
ABORT: Canu 1.7.1
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   unable to assemble uncorrected reads.
ABORT:

can you give me some advance?
Thanks
Molly

**Comment by skoren** [comment]
Are you re-using the same -d option for the 1.7.1 and 1.6 run? That won't work they need to be distinct. It also looks like you're providing subreads which are not corrected as corrected reads, that won't assemble either. Both 1.7.1 and 1.6 are quite old, I would suggest using 1.9 using something like:
```canu -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-raw ../1_data/filtered_subreads.fasta```

**Comment by DongnaMa** [comment]
Thanks, yes, I used same -d option for the 1.7.1 and 1.6 run. I want to know, can I use the 1.9 version to run with the results of the 1.6 version? I don’t want to correct，can I assemble it directly?
------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Sergey Koren"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月28日(星期六) 凌晨0:29
收件人:&nbsp;"marbl/canu"<canu@noreply.github.com&gt;;
抄送:&nbsp;"madongna"<dongnama@foxmail.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [marbl/canu] unable to assemble uncorrected reads (#865)




Are you re-using the same -d option for the 1.7.1 and 1.6 run? That won't work they need to be distinct. It also looks like you're providing subreads which are not corrected as corrected reads, that won't assemble either. Both 1.7.1 and 1.6 are quite old, I would suggest using 1.9 using something like:
 canu -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-raw ../1_data/filtered_subreads.fasta

**Comment by skoren** [comment]
You can use 1.9 or 1.7.1 with the corrected reads from 1.6 by providing the corrected reads output from 1.6, something like:
```
canu -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-corrected Pe.pacbio/Pe.correctedReads.fasta.gz
```

This will run both trimming and assembly, if you already also have trimmed reads and want to use those update the command:
```
canu -assemble -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-corrected Pe.pacbio/Pe.trimmedReads.fasta.gz
```

**Comment by DongnaMa** [comment]
Sorry, due to the special of my data ,At the beginning, I assembled directly without correction.&nbsp;I used the version 1.6 to run to the task bat step. Can I do it directly with the 1.9?


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Sergey Koren"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月28日(星期六) 凌晨5:46
收件人:&nbsp;"marbl/canu"<canu@noreply.github.com&gt;;
抄送:&nbsp;"madongna"<dongnama@foxmail.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [marbl/canu] unable to assemble uncorrected reads (#865)




You can use 1.9 or 1.7.1 with the corrected reads from 1.6 by providing the corrected reads output from 1.6, something like:
 canu -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-corrected Pe.pacbio/Pe.correctedReads.fasta.gz  
This will run both trimming and assembly, if you already also have trimmed reads and want to use those update the command:
 canu -assemble -s spec.txt -p asm -d pe_canu19 genomeSize=1.27g -pacbio-corrected Pe.pacbio/Pe.trimmedReads.fasta.gz

**Comment by skoren** [comment]
You're assembling the raw pacbio data without correction or trimming? That's not going to work well unless the data is recent HiFi and has been through the CCS pipeline to improve accuracy.

You can do it in 1.9 as well, use the same command you did for 1.6 just use a different -d option so the runs don't collide.

**Comment by DongnaMa** [comment]
Sorry, I have another problem. I ran utgovl pipeline was fast when used version 1.6, but utgovl was very slow when used version 1.9. I calculate it will take about 2 months to assemble the 1G genome. Is there any way to make assembly faster?&nbsp;My configuration file is roughly as follows：
--
-- &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (tag)Threads
-- &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(tag)Memory &nbsp; &nbsp; &nbsp; &nbsp; |
-- &nbsp; &nbsp; &nbsp; &nbsp;(tag) &nbsp; &nbsp; &nbsp; &nbsp; | &nbsp; &nbsp; &nbsp; &nbsp; | &nbsp;algorithm
-- &nbsp; &nbsp; &nbsp; &nbsp;------- &nbsp;------ &nbsp;-------- &nbsp;-----------------------------
-- Grid: &nbsp;meryl &nbsp; &nbsp; 50 GB &nbsp; &nbsp;8 CPUs &nbsp;(k-mer counting)
-- Grid: &nbsp;hap &nbsp; &nbsp; &nbsp; 50 GB &nbsp; 24 CPUs &nbsp;(read-to-haplotype assignment)
-- Grid: &nbsp;cormhap &nbsp; 31 GB &nbsp; 16 CPUs &nbsp;(overlap detection with mhap)
-- Grid: &nbsp;obtovl &nbsp; &nbsp;50 GB &nbsp; 16 CPUs &nbsp;(overlap detection)
-- Grid: &nbsp;utgovl &nbsp; &nbsp;16 GB &nbsp; 16 CPUs &nbsp;(overlap detection)
-- Grid: &nbsp;cor &nbsp; &nbsp; &nbsp; 24 GB &nbsp; &nbsp;4 CPUs &nbsp;(read correction)
-- Grid: &nbsp;ovb &nbsp; &nbsp; &nbsp; 50 GB &nbsp; &nbsp;1 CPU &nbsp; (overlap store bucketizer)
-- Grid: &nbsp;ovs &nbsp; &nbsp; &nbsp; 30 GB &nbsp; 30 CPUs &nbsp;(overlap store sorting)
-- Grid: &nbsp;red &nbsp; &nbsp; &nbsp; 20 GB &nbsp; &nbsp;8 CPUs &nbsp;(read error detection)
-- Grid: &nbsp;oea &nbsp; &nbsp; &nbsp; &nbsp;8 GB &nbsp; 30 CPUs &nbsp;(overlap error adjustment)
-- Grid: &nbsp;bat &nbsp; &nbsp; &nbsp; 50 GB &nbsp; 24 CPUs &nbsp;(contig construction with bogart)
-- Grid: &nbsp;cns &nbsp; &nbsp; &nbsp;--- GB &nbsp; &nbsp;8 CPUs &nbsp;(consensus)
-- Grid: &nbsp;gfa &nbsp; &nbsp; &nbsp; 32 GB &nbsp; 16 CPUs &nbsp;(GFA alignment and processing)
--
-- In 'Pe.seqStore', found PacBio reads:
-- &nbsp; Raw: &nbsp; &nbsp; &nbsp; &nbsp;0
-- &nbsp; Corrected: &nbsp;18046215
-- &nbsp; Trimmed: &nbsp; &nbsp;18046215



Thank you




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Sergey Koren"<notifications@github.com&gt;;
发送时间:&nbsp;2019年12月28日(星期六) 凌晨5:56
收件人:&nbsp;"marbl/canu"<canu@noreply.github.com&gt;;
抄送:&nbsp;"madongna"<dongnama@foxmail.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [marbl/canu] unable to assemble uncorrected reads (#865)




You're assembling the raw pacbio data without correction or trimming? That's not going to work well unless the data is recent HiFi and has been through the CCS pipeline to improve accuracy.
 
You can do it in 1.9 as well, use the same command you did for 1.6 just use a different -d option so the runs don't collide.

**Comment by skoren** [comment]
Wasn't your original run using the raw reads and trying to assemble that? I expect it was fast because it wasn't finding any overlaps and it wouldn't have assembled anything. 

The wall time of your assembly is going to depend on how many cores you get assigned on your cluster, are your jobs spending most time waiting for scheduling or running, how long are running jobs taking when they get scheduled? The FAQ has some additional suggestions to speed up overlapping for repetitive genomes.
